Declare ptraddr_t in (sys/)stddef.h and not in stdint.h

brooksdavis opened the pull request:
We need to pick one place for ptraddr_t and I've upstreamed it to stddef.h matching the various CHERI LLVM ports. I've removed the definition from stdint.h and removed the long unused vaddr_t while here.

brooksdavis committed:
remove vaddr_t

It has been deprecated for some time and replaced by ptraddr_t.

brooksdavis committed:
Include (sys/)stddef.h for ptraddr_t defintion

brooksdavis committed:
sys/signal.h: remove is_magic_sighandler_constant

This function is only used in freebsd64_signal.c so move it there.  This
avoids the need for ptraddr_t to be declared in sys/signal.h.

brooksdavis committed:
Don't declare ptraddr_t in stdint.h

Stick to stddef.h like ptrdiff_t.

bsdjhb commented on sys/compat/freebsd64/freebsd64_signal.c:
```diff
@@ -81,6 +81,17 @@ convert_sigevent64(const struct sigevent64 *sig64, struct sigevent *sig)
 }
 
+static inline bool
+is_magic_sighandler_constant(ptraddr_t handler) {
```
Nit, `{` should be on its own line (but that's an old bug it seems)

bsdjhb reviewed (commented):
Presumably the libunwind change needs to cycle through libunwind first?  For ZFS, it is probably best to fix it directly in zfs.git.

brooksdavis commented:
I've pushed with a subrepo pull commit for testing. I'll manually push that bit once CI completes.

brooksdavis commented on sys/contrib/subrepo-openzfs/include/sys/types.h:
```diff
@@ -1,12 +1,7 @@
-typedef size_t     __ptraddr_t;
-#endif
-typedef __ptraddr_t             ptraddr_t;
+typedef	size_t			ptraddr_t;
```
I've added a comment to this effect along with some other required types for now.

brooksdavis committed:
git subrepo pull --force sys/contrib/subrepo-openzfs

subrepo:
  subdir:   "sys/contrib/subrepo-openzfs"
  merged:   "ce3793d09eb6"
upstream:
  origin:   "https://github.com/CTSRD-CHERI/zfs.git"
  branch:   "cheri-hybrid"
  commit:   "ce3793d09eb6"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"
